From e2a8b043257857ac7508c641debecc1e690cdae6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Mon, 28 Sep 2026 21:15:00 +0200
Subject: [PATCH] malloc.c: drop nfences used only in assert

Fails with gcc-16 due to Werror from:
Makefile.palm:CFLAGS := -O2 -g -D_GNU_SOURCE=1 -DPIC -fPIC -Wall -Werror -Wstrict-prototypes \

it wasn't touched in 10 years, so the assert isn't as useful as it
used to be.

malloc.c: In function 'add_segment':
malloc.c:3687:7: error: variable 'nfences' set but not used [-Werror=unused-but-set-variable=]
 3687 |   int nfences = 0;
      |       ^~~~~~~
cc1: all warnings being treated as errors

Upstream-Status: Pending [webOS OSE is no longer maintained on github]
Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 malloc.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/malloc.c b/malloc.c
index 70c0780..8a0ddb7 100644
--- a/malloc.c
+++ b/malloc.c
@@ -3684,7 +3684,6 @@ static void add_segment(mstate m, char* tbase, size_t tsize, flag_t mmapped) {
   msegmentptr ss = (msegmentptr)(chunk2mem(sp));
   mchunkptr tnext = chunk_plus_offset(sp, ssize);
   mchunkptr p = tnext;
-  int nfences = 0;
 
   /* reset top to new space */
   init_top(m, (mchunkptr)tbase, tsize - TOP_FOOT_SIZE);
@@ -3702,13 +3701,11 @@ static void add_segment(mstate m, char* tbase, size_t tsize, flag_t mmapped) {
   for (;;) {
     mchunkptr nextp = chunk_plus_offset(p, SIZE_T_SIZE);
     p->head = FENCEPOST_HEAD;
-    ++nfences;
     if ((char*)(&(nextp->head)) < old_end)
       p = nextp;
     else
       break;
   }
-  assert(nfences >= 2);
 
   /* Insert the rest of old top into a bin as an ordinary free chunk */
   if (csp != old_top) {
